### PR TITLE
Property no unknown

### LINKED ISSUE #4020: Fix false positives for overflowX for CSS-in-JS in property-no-unknown
> Clearly describe the bug

Using React and the CSS object notation leads to an error for the following piece of code:
```jsx
<div style={{overflowX: 'scroll', color: 'black'}}>
```
error:
```shell
demo.jsx
68:41  ×  Unexpected unknown property "overflowx"   property-no-unknown
```
> Which rule, if any, is the bug related to?

`property-no-unknown`

> What CSS is needed to reproduce the bug?

```js
style={overflowX: 'scroll'}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": ["stylelint-config-standard", "stylelint-config-prettier"],
}
```

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI, VScode

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
demo.jsx
68:41  ×  Unexpected unknown property "overflowx"   property-no-unknown
```

### PR DESCRIPTION
**Pull request by vankop**
> Which issue, if any, is this issue related to?

closes #4020 

> Is there anything in the PR that needs further explanation?

nothing

### CODE REVIEW

File: lib/rules/property-no-unknown/__tests__/index.js [hunk 1]
```diff
@@ -229,6 +229,16 @@ testRule(rule, {
...
+    },
+    {
+      code: `
+        import glm from 'glamorous';
+        const Component = glm.a({
+            overflowX: 'auto',
+            overflowWrap: 'auto',
+            flexDirection: 'row',
+        });
+      `
```
Comment (evilebottnawi) [inline lib/rules/property-no-unknown/__tests__/index.js]: Can we add test with failed example?

Comment (vankop) [reply lib/rules/property-no-unknown/__tests__/index.js]: done

**Comment by vankop** [comment]
Don't have we a problem with node 8 runner in CI? always fails in node 8 runner

**Comment by evilebottnawi** [comment]
Strange error, agree

**Comment by vankop** [comment]
As I understand we using windows runners, so we can use build-in images for runners, see https://www.appveyor.com/docs/windows-images-software/#node-js instead of installing node 8 manually https://github.com/stylelint/stylelint/blob/master/appveyor.yml#L19 . I am not familiar with appveyor CI/CD, but it is pretty common functionality + if we will add lock file we could install packages faster with `npm ci`

Also one idea, we using cache from `node_modules`, but I don't know is it same cache for node-8 and node-10, if it is so it could lead to some uncommon problems, for example, node-sass build is different depends on platform, see https://github.com/sass/node-sass/blob/master/package.json

**Comment by evilebottnawi** [comment]
> As I understand we using windows runners, so we can use build-in images for runners, see https://www.appveyor.com/docs/windows-images-software/#node-js instead of installing node 8 manually https://github.com/stylelint/stylelint/blob/master/appveyor.yml#L19

This will not add much time to us, but will increase the maintenance time on the project

> I am not familiar with appveyor CI/CD, but it is pretty common functionality + if we will add lock file we could install packages faster with npm ci

/cc @stylelint/core what is blocker for using lock files, it is good practice

**Comment by vankop** [comment]
> This will not add much time to us, but will increase the maintenance time on the project

Why it will be increased? AppVeyor is pretty popular and they maintain there images, so it will make behavior of our code *expected* since they create node 8 environment for us instead of using combo => unknown runner env + node 8 (If I understand how appveyor works 😁) For gitlab it is common practice

**Comment by evilebottnawi** [comment]
> Why it will be increased?

Who will update the images? Tomorrow we may be busy and not find it at this time

**Comment by vankop** [comment]
Looks like we have misunderstanding each other.

How it works now:
1. Appveyor starts default image on virtual machine
2. clone repo in it
3. cd in repo root
4. Runs `install` commands: a. Start powershell install node 8 b. runs `npm install` https://github.com/stylelint/stylelint/blob/master/appveyor.yml
5. runs tests

Cost of maintenance each version add new environment

How it will work with predefined images:
1. Appveyor starts **node 8 image** on virtual machine
2. clone repo in it
3. cd in repo root
4. Runs `install` commands: ~~a. Start powershell install node 8~~ b. runs `npm install` https://github.com/stylelint/stylelint/blob/master/appveyor.yml
5. runs tests

example config:
```
environment:
    matrix:
        - image: Node.js 8.16.0
        - image: Node.js 10.16.0
```

Cost of maintenance each version add new image in list

**Comment by vankop** [comment]
If I make mistake in describing it will be great, if you will clarify how it works for me

**Comment by hudochenkov** [comment]
> -   Fixed: `property-no-unknown` for overflowX for CSS-in-JS ([#4184](https://github.com/stylelint/stylelint/pull/4184)).

**Pull request merged**